Utils/Extensions/NSURLExtensions.swift

```diff
@@ -236,6 +236,17 @@ extension NSURL {
     public var isLocal: Bool {
         return host?.lowercaseString == "localhost" || host == "127.0.0.1" || host == "::1"
     }
+    
+    /**
+     Returns whether the URL's scheme is one of those listed on the official list of URI schemes found here: http://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
+     This only accepts historical and permanent schemes: provisional schemes are not accepted.
+     */
+    public var schemeIsValid: Bool {
+        let historicalSchemes = ["fax", "filesystem", "mailserver", "modem", "pack", "prospero", "snews", "videotex", "wais", "z39.50"]
+        let permanentSchemes = ["aaa", "aaas", "about", "acap", "acct", "cap", "cid", "coap", "coaps", "crid", "data", "dav", "dict", "dns", "example", "file", "ftp", "geo", "go", "gopher", "h323", "http", "https", "iax", "icap", "im", "imap", "info", "ipp", "ipps", "iris", "iris.beep", "iris.lwz", "iris.xpc", "iris.xpcs", "jabber", "ldap", "mailto", "mid", "msrp", "msrps", "mtqp", "mupdate", "news", "nfs", "ni", "nih", "nntp", "opaquelocktoken", "pkcs11", "pop", "pres", "reload", "rtsp", "rtsps", "rtspu", "service", "session", "shttp", "sieve", "sip", "sips", "sms", "snmp", "soap.beep", "soap.beeps", "stun", "stuns", "tag", "tel", "telnet", "tftp", "thismessage", "tip", "tn3270", "turn", "turns", "tv", "urn", "vemmi", "vnc", "ws", "wss", "xcon", "xcon-userid", "xmlrpc.beep", "xmlrpc.beeps", "xmpp", "z39.50r", "z39.50s"]
+        
+        return (historicalSchemes + permanentSchemes).contains(scheme)
```

Comment:
Let's extract this result to a private constant outside `schemeIsValid` so we don't have to generate this list each time it's called.